### DISCUSSION TITLE
Optimizing AI Inference Latency using Serverless Functions (AWS Lambda/GCP Functions)

### DISCUSSION THREAD

**Discussion by MedAymenChakroun** [ahlem-phantom/AI-HealthCare-Assistant#19] [Q&A]
Currently, the Flask server handles the ML models. If we were to migrate the symptom detection and facial recognition to a Serverless (FaaS) architecture to save costs, how would we deal with the 'Cold Start' problem caused by loading heavy ML libraries (like TensorFlow or Scikit-Learn)?

I'm curious if we should use Containerized Lambdas or if there is a way to optimize the model weights to make them light enough for instant cloud execution without sacrificing the accuracy of the illness detection
